### PR TITLE
sql: remove the planner ref from planNodes

### PR DESCRIPTION
**Pull request by knz**
The first commit from #20169.

Visible changes: the `*planner` reference is removed from planNodes.

Release notes: none.

### CODE REVIEW

**Comment by knz** [comment]
cc @jordanlewis

**Comment by knz** [comment]
@justinj I was advised by Vivek that you could help @jordanlewis with reviews around this area.

**Comment by jordanlewis** [comment]
Haven't reviewed carefully, but blanket +1 on removing references to `planner` from plan nodes. I suggest limiting the scope of this PR to exactly that if you have not already.

**Comment by knz** [comment]
ok they're all gone.

**Comment by knz** [comment]
Rebased on top of #20169 so no file renames, for your reviewing pleasure.

File: pkg/sql/group.go [hunk 1]
```diff
@@ -484,7 +481,12 @@ func (n *groupNode) addIsNotNullFilter(where *filterNode, render *renderNode) {
 //
 // We only have a desired ordering if we have a single MIN or MAX aggregation
 // with a simple column argument and there is no GROUP BY.
-func (n *groupNode) desiredAggregateOrdering() sqlbase.ColumnOrdering {
+//
+// TODO(knz/radu): it's expensive to instantiate the aggregate
+// function here just o determine whether it's a min or max. We could
```
Comment (justinj) [inline pkg/sql/group.go]: `o -> to`

File: pkg/sql/render.go [hunk 2]
```diff
@@ -295,14 +293,14 @@ func (p *planner) SelectClause(
 }
 
 // initFrom initializes the table node, given the parsed select expression
-func (r *renderNode) initFrom(
-	ctx context.Context, parsed *tree.SelectClause, scanVisibility scanVisibility,
+func (p *planner) initFrom(
```
Comment (justinj) [inline pkg/sql/render.go]: any particular reason this and other methods moved from the renderNode to the planner?

**Review by justinj — APPROVED** [review]
General idea LGTM, I have some other questions that I messaged you about, but definitely in favour of this decoupling!

**Comment by knz** [comment]
There's an error in the last commit. Fixing now.

---

Review status: 0 of 35 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/group.go, line 486 at r16](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154#-KzOcM0jrwLhpQeRxcPO-r16-486:-KzUD5aqN7FKLwCGKyui:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/41e503b20401e26ae118f64b6f1514c34c1539ba/pkg/sql/group.go#L486)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

`o -> to`
</blockquote></details>

Fixed.

---

*[pkg/sql/render.go, line 296 at r16](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154#-KzOcM0kxygUzEUfMhdT-r16-296:-KzUD9_wjJtfMoNCJoJM:b-4jwlc) ([raw file](https://github.com/cockroachdb/cockroach/blob/41e503b20401e26ae118f64b6f1514c34c1539ba/pkg/sql/render.go#L296)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

any particular reason this and other methods moved from the renderNode to the planner?
</blockquote></details>

This function calls `(*planner).getSources()` to construct the logical plan corresponding to the FROM clause. The `planner` reference has to come from somewhere. Si it's either making it a method of `*planner` or passing `p*planner`  as explicit argument. 

Now consider the name "planner". The planner object has too many responsibilities right now, but its name is clear: its purpose, the only purpose it *eventually* will fulfill, is to take an AST and construct a logical plan. So a method that takes a FROM AST and returns a logical plan (or, in this case, annotates it in a pre-initialized logical plan node) is the proper design.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154#-:-KzUDnpRCsYnVUPgLGmq:b-1j6x2s)*

**Comment by knz** [comment]
I had forgotten a step because of some oddity in the distsql physical planner. Added an intermediate commit that addresses the problem. The PR should be ready now. PTAL. 

---

Review status: 0 of 45 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154#-:-KzWExBV8K2hvAxtKb3p:bca5bfa)*

File: pkg/sql/trace.go [hunk 3]
```diff
@@ -43,9 +42,9 @@ type traceNode struct {
 	// around the interaction of SQL with KV. Some of the messages are per-row.
 	kvTracingEnabled bool
 
-	// recording is set if this node started tracing on the session. If it did,
-	// then Close() needs to stop the recording.
-	recording bool
+	// stopTracing is set is set if this node started tracing on the
```
Comment (RaduBerinde) [inline pkg/sql/trace.go]: is set is set. Also "Close() must close it" is confusing

Why is this a function? It seems that the only purpose is so we can implicitly store `params.p.session` in it (so we're still storing runtime state, we just "hide" it a bit).

**Comment by RaduBerinde** [comment]
LGTM

**Comment by knz** [comment]
Review status: 0 of 46 files reviewed at latest revision, 3 unresolved discussions.

---

*[pkg/sql/trace.go, line 45 at r23](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154#-KzOcM0mxRZAT9iAmvW7-r23-45:-KztgonGcXLv3MWgVzCS:b-aig6ka) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ab209691caa92dd2dc4fdbdff6c6c5552a26fc5/pkg/sql/trace.go#L45)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

is set is set. Also "Close() must close it" is confusing

Why is this a function? It seems that the only purpose is so we can implicitly store `params.p.session` in it (so we're still storing runtime state, we just "hide" it a bit).
</blockquote></details>

It reduces the range of operations that can be performed on the stored session. This way, the only thing the planNode can do is stopping the tracing. Keeping the planner/session around would leave the door open to arbitrary other things.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20154)*

**Comment by knz** [comment]
TFYRs

**Pull request merged**